consumer_service/consumer_service.go

```diff
@@ -0,0 +1,81 @@
+package consumer_service
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+type FileStorage interface {
+	Add(endpoint string, dealID uint64, dataHash string, data []byte) error
+}
+
+var _ FileStorage = &ConsumerServiceFileStorage{}
+
+type ConsumerServiceFileStorage struct {
+	oraclePrivKey *btcec.PrivateKey
+}
+
+func NewConsumerService(oraclePrivKey *btcec.PrivateKey) FileStorage {
+	return &ConsumerServiceFileStorage{
+		oraclePrivKey: oraclePrivKey,
+	}
+}
+
+func (s *ConsumerServiceFileStorage) Add(endpoint string, dealID uint64, dataHash string, data []byte) error {
+	dataUrl := endpoint + "/v0/data/" + strconv.FormatUint(dealID, 10) + "/" + dataHash
```

Comment:
the url path has been changed as https://github.com/medibloc/panacea-dep-consumer/pull/1/files#diff-72235548d0574fdd05a674ff9561de593400b03566caf7fa69cbd14add25f417R26